Recruit-Training/Embedded-Recruit-Training/Core/Src/main.c

```diff
@@ -0,0 +1,458 @@
+    RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
+    RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
+    RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV8;
+    RCC_ClkInitStruct.APB2CLKDivider = RCC_HCLK_DIV4;
+
+    if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_5) != HAL_OK)
+    {
+        Error_Handler();
+    }
+}
+
+/**
+  * @brief CAN2 Initialization Function
+  * @param None
+  * @retval None
+  */
+static void MX_CAN2_Init(void)
+{
+
+    /* USER CODE BEGIN CAN2_Init 0 */
+
+    /* USER CODE END CAN2_Init 0 */
+
+    /* USER CODE BEGIN CAN2_Init 1 */
+
+    /* USER CODE END CAN2_Init 1 */
+    hcan2.Instance = CAN2;
+    hcan2.Init.Prescaler = 4;
+    hcan2.Init.Mode = CAN_MODE_NORMAL;
+    hcan2.Init.SyncJumpWidth = CAN_SJW_1TQ;
+    hcan2.Init.TimeSeg1 = CAN_BS1_5TQ;
+    hcan2.Init.TimeSeg2 = CAN_BS2_4TQ;
+    hcan2.Init.TimeTriggeredMode = DISABLE;
+    hcan2.Init.AutoBusOff = DISABLE;
+    hcan2.Init.AutoWakeUp = DISABLE;
+    hcan2.Init.AutoRetransmission = DISABLE;
+    hcan2.Init.ReceiveFifoLocked = DISABLE;
+    hcan2.Init.TransmitFifoPriority = DISABLE;
+
+    if (HAL_CAN_Init(&hcan2) != HAL_OK)
+    {
+        Error_Handler();
+    }
+
+    /* USER CODE BEGIN CAN2_Init 2 */
+
+    /* USER CODE END CAN2_Init 2 */
+
+}
+
+/**
+  * @brief GPIO Initialization Function
+  * @param None
+  * @retval None
+  */
+static void MX_GPIO_Init(void)
+{
+    GPIO_InitTypeDef GPIO_InitStruct = {0};
+
+    /* GPIO Ports Clock Enable */
+    __HAL_RCC_GPIOH_CLK_ENABLE();
+    __HAL_RCC_GPIOA_CLK_ENABLE();
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+
+    /*Configure GPIO pin Output Level */
+    HAL_GPIO_WritePin(GPIOA, LED_RED_Pin | LED_GREEN_Pin | GPIO_PIN_7, GPIO_PIN_RESET);
+
+    /*Configure GPIO pins : LED_RED_Pin LED_GREEN_Pin PA7 */
+    GPIO_InitStruct.Pin = LED_RED_Pin | LED_GREEN_Pin | GPIO_PIN_7;
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+}
+
+/* USER CODE BEGIN 4 */
+void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
+{
+    CAN_RxHeaderTypeDef hdr;
+    uint8_t             data[8] = {0};
+
+    if (HAL_CAN_GetRxMessage(hcan, CAN_RX_FIFO0, &hdr, data) != HAL_OK)
+    {
+        return;
+    }
+
+    //TODO: Match StdId of header and data length content for green and blue messages and check data for set bit of toggling green and blue led
+    if (hdr.StdId == BLUE_MESSAGE_STDID && hdr.DLC == 1) 
+    {
+        uint8_t bit = 0b10001001
+
+        if(data[0] & bit == bit)  // & operator is  the AND gate to multiply 8-bit numbers
+        {
+            BlueToggleLED = 1;
+        }
+
+        else
+        {    
+            BlueToggleLED = 0;
+        }
+        
+    }
+
+    if(hdr.StdId == GREEN_MESSAGE_STDID && hdr.DLC == 1)
+    {
+        uint8_t mask = 0b00000011
+
+        if(data[0] & mask == mask)
```

Comment:
The instructions say that only bit 0 and bit 1 should be set. With this code, if data[0] is `0b11111111`, it would still set the toggle flag to 1, which is incorrect